packages/admin/blocks-admin/src/blocks/factories/createListBlock.tsx

```diff
@@ -208,7 +218,12 @@ export function createListBlock<T extends BlockInterface, AdditionalItemFields e
                 adminMeta: { route: previewCtx.parentUrl },
             };
         },
-        isValid: async (state) => parallelAsyncEvery(state.blocks, async (c) => block.isValid(c.props)),
+        isValid: async (state) =>
+            parallelAsyncEvery(
+                state.blocks,
+                async (c) =>
+                    block.isValid(c.props) && (minVisibleBlocks ? minVisibleBlocks <= state.blocks.filter((block) => block.visible).length : true),
```

Comment:
The minVisibleBlocks check is performed multiple times. It would be better if it would only be checked once. Maybe something like this?

```ts
isValid: async (state) => Promise.all([parallelAsyncEvery(state.blocks, async (c) => block.isValid(c.props)), minVisibleBlocks ? minVisibleBlocks <= state.blocks.filter((block) => block.visible).length : true])
```

Reply:
`parallelAsyncEvery` does a `Promise.all` internally 🤔

Reply:
The check `minVisibleBlocks ? minVisibleBlocks <= state.blocks.filter((block) => block.visible).length : true` is currently performed `state.blocks.length` n times, while it is only necessary once. So it needs to be outside of the parallelAsyncEvery call.

Reply:
Ah now I get it thanks. I had to rewrite it a bit because it has to evaluate to one boolean :) Here is the commit [be7c8ac](https://github.com/vivid-planet/comet/pull/1865/commits/be7c8ac261512c45257dfd99cf7d55a6c5339d27)